src/Executor/Setup.cs

```diff
@@ -0,0 +1,290 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.DotNet.Execute
+{
+    public class Setup
+    {
+        public Dictionary<string, Setting> Settings { get; set; }
+        public Dictionary<string, Command> Commands { get; set; }
+        public Dictionary<string, Tool> Tools { get; set; }
+        public Dictionary<string, string> SettingParameters { get; set; }
+        public string Os { get; set; }
+        public string ConfigurationFilePath { get; set; }
+
+
+        private string FindSettingValue(string valueToFind)
+        {
+            if (Settings.ContainsKey(valueToFind))
+            {
+                return Settings[valueToFind].DefaultValue;
+            }
+            return null;
+        }
+
+        private string FindSettingType(string valueToFind)
+        {
+            if (Settings.ContainsKey(valueToFind))
+            {
+                return Settings[valueToFind].ValueType;
+            }
+            return null;
+        }
+        
+        public void prepareValues(string os, Dictionary<string, string> parameters, string configFile)
+        {
+            SettingParameters = new Dictionary<string, string>(parameters);
+            Os = os;
+            ConfigurationFilePath = configFile;
+        }
+
+        public int ExecuteCommand(string commandSelectedByUser)
+        {
+            string[] commandToRun = BuildCommand(commandSelectedByUser);
+            if(commandToRun != null)
+            {
+                Console.ForegroundColor = ConsoleColor.DarkYellow;
+                Console.WriteLine("Running: {0} {1}", commandToRun[0], commandToRun[1]);
+                Console.ResetColor();
+
+                Run runCommand = new Run();
+                int result = runCommand.ExecuteProcess(commandToRun[0], commandToRun[1]);
+                if (result == 0)
+                {
+                    Console.ForegroundColor = ConsoleColor.Green;
+                    Console.WriteLine("Build Succeeded.");
+                }
+                else
+                {
+                    Console.ForegroundColor = ConsoleColor.Red;
+                    Console.WriteLine("Build Failed.");
+                }
+                Console.ResetColor();
+
+                return result;
+            }
+            return 1;
+        }
+
+        public string[] BuildCommand(string commandSelectedByUser, Dictionary<string, string> parameters = null)
```

Comment:
Magic arrays (where each index has special meaning) are difficult to understand when reading code, since the "members" aren't self-documenting. You could actually return a "ProcessStartInfo" from this method. :-)